core/migration/safestore.go

```diff
@@ -0,0 +1,36 @@
+package migration
+
+import (
+	"context"
+	"sync"
+
+	"github.com/anyproto/anytype-heart/pkg/lib/database"
+	"github.com/anyproto/anytype-heart/pkg/lib/localstore/objectstore"
+)
+
+type QueryableStore interface {
+	Query(q database.Query) (records []database.Record, err error)
+	Lock()
+}
+
+type safeStore struct {
+	store  objectstore.ObjectStore
+	locked bool
+	m      sync.RWMutex
+}
+
+func (s *safeStore) Query(q database.Query) (records []database.Record, err error) {
+	s.m.RLock()
+	if s.locked {
+		s.m.RUnlock()
+		return nil, context.Canceled
+	}
+	s.m.RUnlock()
```

Comment:
Can be in defer statement

Reply:
I did not put Unlock to defer statement, because I don't want to wait until Query will be done. Lock is used here only to access bool value, but Query could be called by multiple threads simulteneously